bcrypt.js

```diff
@@ -9,31 +9,43 @@ var crypto = require('crypto');
 
 /// generate a salt (sync)
 /// @param {Number} [rounds] number of rounds (default 10)
+/// @param {Buffer} [seed] random seed (default crypto.randomBytes(16))
 /// @return {String} salt
-module.exports.genSaltSync = function(rounds) {
+module.exports.genSaltSync = function(rounds, seed) {
     // default 10 rounds
     if (!rounds) {
         rounds = 10;
     } else if (typeof rounds !== 'number') {
         throw new Error('rounds must be a number');
     }
-
-    return bindings.gen_salt_sync(rounds, crypto.randomBytes(16));
+    // seed default is random data
+    if (!seed) {
+	seed = crypto.randomBytes(16);
+    } else if (!Buffer.isBuffer(seed) || seed.length != 16) {
+        throw new Error('seed must be a 16 byte Buffer');
+    }
+    return bindings.gen_salt_sync(rounds, seed);
 };
 
 /// generate a salt
 /// @param {Number} [rounds] number of rounds (default 10)
+/// @param {Buffer} [seed] random seed (default crypto.randomBytes(16))
 /// @param {Function} cb callback(err, salt)
-module.exports.genSalt = function(rounds, ignore, cb) {
+module.exports.genSalt = function(rounds, seed, cb) {
     // if callback is first argument, then use defaults for others
     if (typeof arguments[0] === 'function') {
         // have to set callback first otherwise arguments are overriden
         cb = arguments[0];
-        rounds = 10;
+        rounds = undefined;
     // callback is second argument
     } else if (typeof arguments[1] === 'function') {
         // have to set callback first otherwise arguments are overriden
         cb = arguments[1];
+	seed = undefined;
```

Comment:
Indentation here too, and there are more tabs on later lines. Maybe search for them or use a tool of some form that finds whitespace problems.